### PR TITLE
Adding intellicard functionality.

### PR DESCRIPTION
**Pull request by ScarKy0**
## About the PR

Intellicards can now be used to remove/insert the current AI from the core.
Slight intellicard sprite update.

## Why / Balance

Killing AI makes sense... I think

## Technical details

The brain slot on intellicards and ai core is unlocked but cannot eject, making intellicards swap the brain between itself and the core.
AI Eye gets deleted/created on adding or removing ai from the core to fix a bug where the AI Eye would stay visible forever after being removed.
InstrinsicUI now removes all actions it granted on shutdown. Seeing it was only used on AI and Aghost and after some testing it seems to be working all fine.

## Media

https://github.com/user-attachments/assets/e8cd9280-1954-448e-b091-9234387bfc6b

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

InstrinsicUI now removes all actions it granted on shutdown. Unsure if this is a breaking change or not.

**Changelog**

:cl: ScarKy0, Fildrance
- add: Using an intellicard on the AI core now swaps the AI between the core and intellicard. (You can evac with the AI!)
- add: Added an intellicard to the RD's locker.

### CODE REVIEW

**Comment by ScarKy0** [comment]
If someone knows how to do anything in the TODO please tell me because I don't 😭

**Comment by deltanedas** [comment]
> Make swapping the brain a doAfter.

i think ItemSlot has a field for this that rmc added

**Comment by ScarKy0** [comment]
> i think ItemSlot has a field for this that rmc added

From what ive seen ItemSlotsComponent doesnt have a field for doAfter, unless im missing something

**Comment by ScarKy0** [comment]
God bless fildrance

**Comment by superjj18** [comment]
Just need to make ai cores buildable and anchorable and then AI’s can successfully be revived after being disconnected from the station.

**Comment by AverageNotDoingAnythingEnjoyer** [comment]
Maybe it shouldn't spawn in RD locker for now, as AI still doesn't have ways to protect itself against an antag breaking in and carding them

File: Content.Shared/Silicons/StationAi/SharedStationAiSystem.cs [hunk 1]
```diff
@@ -340,6 +347,13 @@ private void OnAiRemove(Entity<StationAiCoreComponent> ent, ref EntRemovedFromCo
     {
         if (_timing.ApplyingState)
             return;
+        
+
+        if (TryComp(args.Entity, out ActionsComponent? comp))
+        {
+            comp.Actions.Clear();
+            Dirty(args.Entity, comp);
+        }
```
Comment (metalgearsloth) [inline Content.Shared/Silicons/StationAi/SharedStationAiSystem.cs:356]: Pretty sure actionsystem has a method for this like clearactions.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Remove the PAI stuff for now I reckon.

**Comment by ScarKy0** [comment]
Well shit its all on fire

**Comment by slarticodefast** [comment]
Tested in-game and works as intended.

**Comment by superjj18** [comment]
Can we perhaps remove mapped intellicards and instead put one in RD's locker?

**Comment by ScarKy0** [comment]
> Make the intelicard require a lenghty do-after like disarming the nuke does. That way someone can't just suicide rush into AI chamber and intelicard them away, and also provides a MALF AI or just regular AI in general the chance to warn crew or fight back before they are completely neutralized. Perhaps even make it tell the AI "hey someone is uploading you into an Intelicard!"

Thats the plan!
I will be trying to do something similiar when I continue working on intellicards after this, for now this serves as a basic way to be able to kill the AI

> Can we perhaps remove mapped intellicards and instead put one in RD's locker?

There is one in RD's locker but I know nothing of mapping to remove the mapped ones

**Pull request merged**